packages/ssz/src/view/stableContainer.ts

```diff
@@ -0,0 +1,250 @@
+            }
+            return;
+          }
+          const leafNodePrev = getNodeAtDepth(this.node, this.type.depth, index) as LeafNode;
+          const leafNode = leafNodePrev.clone();
+          fieldType.tree_setToNode(leafNode, value);
+          this.tree.setNodeAtDepth(this.type.depth, index, leafNode);
+          // only update the active field if necessary
+          if (!this.type.tree_getActiveField(this.tree.rootNode, index)) {
+            this.tree.rootNode = this.type.tree_setActiveField(this.tree.rootNode, index, false);
+          }
+        },
+      });
+    }
+
+    // If the field type is composite, the value to get and set will be another TreeView (if not nullish). The parent TreeView must
+    // cache the view itself to retain the caches of the child view. To set a value the view must return a node to
+    // set it to the parent tree in the field gindex.
+    else if (isCompositeType(fieldType)) {
+      Object.defineProperty(CustomContainerTreeView.prototype, fieldName, {
+        configurable: false,
+        enumerable: true,
+
+        // Returns TreeView of fieldName
+        get: function (this: CustomContainerTreeView) {
+          const gindex = toGindexBitstring(this.type.depth, index);
+          const subtree = this.tree.getSubtree(gindex);
+          if (optional && this.type.tree_getActiveField(this.tree.rootNode, index) === false) {
+            return null;
+          }
+          return fieldType.getView(subtree);
+        },
+
+        // Expects TreeView of fieldName
+        set: function (this: CustomContainerTreeView, value: unknown) {
+          if (optional && value == null) {
+            this.tree.setNodeAtDepth(this.type.depth, index, zeroNode(0));
+            // only update the active field if necessary
+            if (this.type.tree_getActiveField(this.tree.rootNode, index)) {
+              this.tree.rootNode = this.type.tree_setActiveField(this.tree.rootNode, index, false);
+            }
+            return;
+          }
+          const node = fieldType.commitView(value);
+          this.tree.setNodeAtDepth(this.type.depth, index, node);
+          // only update the active field if necessary
+          if (!this.type.tree_getActiveField(this.tree.rootNode, index)) {
+            this.tree.rootNode = this.type.tree_setActiveField(this.tree.rootNode, index, false);
+          }
+        },
+      });
+    }
+
+    // Should never happen
+    else {
+      /* istanbul ignore next - unreachable code */
+      throw Error(`Unknown fieldType ${fieldType.typeName} for fieldName ${fieldName}`);
+    }
+  }
+
+  // Change class name
+  Object.defineProperty(CustomContainerTreeView, "name", {value: type.typeName, writable: false});
+
+  return CustomContainerTreeView as unknown as ContainerTreeViewTypeConstructor<Fields>;
+}
+
+type BytesRange = {start: number; end: number};
+
+/**
+ * Precompute fixed and variable offsets position for faster deserialization.
```

Comment:
tweak comment for this type nothing that this should be computed once per value instead of once per type like in Container